src/Misc/layoutbin/update.sh.template

```diff
@@ -172,6 +176,59 @@ if [[ "$currentplatform" == 'darwin'  && restartinteractiverunner -eq 0 ]]; then
     fi
 fi
 
+# NODE 16: DUPLICATED THE ABOVE LOGIC FOR NODE16
+# fix upgrade issue with macOS when running as a service
+attemptedtargetedfix=0
+currentplatform=$(uname | awk '{print tolower($0)}')
+if [[ "$currentplatform" == 'darwin'  && restartinteractiverunner -eq 0 ]]; then
```

Comment:
should we determine if it was started with node 12 or node 16 and only run the code for that version?

Reply:
I felt vary of introducing more logic into an already complicated process. There's of course the advantage of your approach making it easier to introduce node18 in the future :)

I was thinking it's a bit easier to remove the old logic once we're done with the migration. There's already the assumption that there's only one runner service active so I didn't see a harm in duplicating it like this (should probably put it in a function at least though).